src/components/app/UserList/ListTable.tsx

```diff
@@ -1,209 +1,198 @@
-import { Avatar, Button, Popconfirm, Table } from 'antd';
+import { LockTwoTone, UnlockTwoTone } from '@ant-design/icons';
+import { Avatar, Button, notification, Popconfirm, Table } from 'antd';
 import { ColumnType } from 'antd/lib/table';
-import Userlist from '../../../types/user/userList';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import updateUserDetails from '../../../apis/update-user-details';
+import UserDetails from '../../../types/user/userDetails';
 
 interface Props {
-  data: Userlist[];
+  onActivate: () => void;
+  data: Partial<UserDetails>[];
 }
 
-const columns: ColumnType<Userlist>[] = [
-  {
-    title: '',
-    dataIndex: 'profileImage',
-    key: 'profileImage',
-    align: 'center',
-    width: '5%',
-    render: (text, record) =>
-      record.profileImage ? (
-        <Avatar src={record.profileImage} />
-      ) : (
-        <Avatar
-          style={{
-            backgroundColor: `#${Math.floor(Math.random() * 16777215).toString(
-              16,
-            )}`,
-          }}
-        >
-          {record.name.slice(0, 1).toUpperCase()}
-        </Avatar>
-      ),
-  },
-  {
-    title: 'Teacher Id',
-    dataIndex: 'teacherId',
-    key: 'teacherId',
-    sorter: (a, b) => {
-      return a.teacherId > b.teacherId ? 1 : -1;
+const ListTable = (props: Props) => {
+  const [loading, setLoading] = useState<boolean>(false);
+  const navigate = useNavigate();
+
+  const updateStatus = async (details: Partial<UserDetails>) => {
+    setLoading(true);
+    try {
+      const res = await updateUserDetails({
+        id: details.id,
+        isActive: !details.isActive,
+      });
+      if (res.data && res.ok) {
+        // eslint-disable-next-line no-console
+        console.log(res.data);
+        notification.success({ message: 'Status Updated Successfully.' });
+      }
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.log(err);
+      notification.error({ message: 'Something went wrong!' });
+    }
+    setLoading(false);
+    props.onActivate();
```

Comment:
@Prodip-Kumar-Paul is this always Activating. Maybe it is deactivating. So I don't think this variable is perfect.

Reply:
@Bijoy-007 nice catch 
update in a min